shared/wallets/routes.tsx

```diff
@@ -64,29 +72,58 @@ class WalletsSubNav extends React.PureComponent<NavigationViewProps<any>> {
   }
 }
 
+const WalletsSubNavigator = createNavigator(
+  WalletsSubNav,
+  StackRouter(Shim.shim(walletsSubRoutes), {initialRouteName: 'wallet'}),
+  {}
+)
+const OnboardingOrWalletsNavigator = createSwitchNavigator(
+  {
+    onboarding: RoutedOnboarding,
+    wallet: WalletsSubNavigator,
+  },
+  {initialRouteName: 'onboarding'}
+)
+
+class _OnboardingOrWallets extends React.Component<any> {
+  static router = OnboardingOrWalletsNavigator.router
+  static navigationOptions = ({navigation}) => {
+    return {
+      header: undefined,
+      headerExpandable: true,
+      // index 0 means we're on the onboarding page, so hide the header
+      headerMode: navigation.state.index === 0 ? 'none' : undefined,
+      headerRightActions: require('./nav-header/container').HeaderRightActions,
+      headerTitle: require('./nav-header/container').HeaderTitle,
+      title: 'Wallet',
+    }
+  }
+
+  componentDidMount() {
+    if (this.props.acceptedDisclaimer) {
+      this.props.navigation.navigate('wallet')
```

Comment:
I'm not sure if it'd make more sense to dispatch one of our navigation actions here instead of using `navigate` directly?

Reply:
i think this is fine. its a router thing natively so it can use the native stuff